core/client/kt/requests.go

```diff
@@ -75,7 +77,7 @@ func (v *Verifier) CreateUpdateEntryRequest(
 	if err != nil {
 		return nil, err
 	}
-	previous := objecthash.ObjectHash(getResp.GetLeafProof().GetLeaf().GetLeafValue())
+	previous := objecthash.ObjectHash(prevEntry)
```

Comment:
Please add a TODO to change this hash to plain SHA256. 
The previous entry is stored as `[]byte` and therefore we should hash it directly.